### PR TITLE
Add a UI option to graceful_shutdown

### PR DESCRIPTION
**Pull request by nmahendru**
When the operator is done using the HSM, they should have
an option to gracefully shutdown the machine.
That should prevent:
1. File system corruption.
2. May aid in longevity of the HSM itself.

This PR adds an option to the prompt and executed the linux shutdown
command on the back.

### CODE REVIEW

File: java/gui/src/main/java/com/squareup/subzero/SystemShutdown.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+public class SystemShutdown {
+    /**
+     * Initiate a graceful shutdown of the machine(Linux only, NOP for others(Mac)).
+     *
+     * An attempt is made to check if the hardserver is running on the machine.
+     *  - If yes, then really shutdown
+     *  - If not, it's a NOP. We probably don't want to shutdown while testing.
```
Comment (syncom) [inline java/gui/src/main/java/com/squareup/subzero/SystemShutdown.java:12]: I like the idea of having a distinction between "real" and (unit) "testing" cases. I was wondering if this logic also means if the hardserver crashes for some reason on the "real" machine, we would not be shown a graceful shutdown option.

Comment (nmahendru) [reply java/gui/src/main/java/com/squareup/subzero/SystemShutdown.java:12]: The hardserver is configured by default to restart immediately on shutdown. Considering we control the number of calls we do, we always start up the machine for use(clear ram and no lingering connnections, states) and this is not a network appliance, and we always test the whole software stack before a release, I believe the crash is highly unlikely.

File: java/gui/src/main/java/com/squareup/subzero/framebuffer/Screens.java [hunk 2]
```diff
@@ -195,9 +195,10 @@ public String promptPasswordChange() throws IOException {
     return promptPassword("Please confirm the password");
   }
 
-  public enum ExitOrRestart {
+  public enum DisplayReturn {
```
Comment (nmahendru) [inline java/gui/src/main/java/com/squareup/subzero/framebuffer/Screens.java:198]: renaming this to make a little more sense. Better name suggestions welcome.

Comment (syncom) [reply java/gui/src/main/java/com/squareup/subzero/framebuffer/Screens.java:198]: Would `ExitOrRestartOrPoweroff` be better (for the explicitness)?

**Comment by syncom** [comment]
nit: Typo: the word `gracefull` should be `graceful` in the PR title and in the commit message.

File: java/gui/src/main/java/com/squareup/subzero/framebuffer/Screens.java (hunk 2 above)
Comment (nmahendru) [reply java/gui/src/main/java/com/squareup/subzero/framebuffer/Screens.java]: done.

**Comment by nmahendru** [comment]
> nit: Typo: the word `gracefull` should be `graceful` in the PR title and in the commit message.

resolved.

File: java/gui/src/main/java/com/squareup/subzero/framebuffer/Screens.java [hunk 3]
```diff
@@ -207,12 +208,13 @@ public String promptPasswordChange() throws IOException {
...
    */
-  public ExitOrRestart displayQRCode(String encodedData) throws IOException {
+  public ExitOrRestartOrPowerOff displayQRCode(String encodedData) throws IOException {
 
     System.out.println("Displaying QR code. Data: " + encodedData);
 
-    String message = "We are done. Please scan the following QR-code with the blue scanner.\n" +
-        "Then type 'exit' + <enter> or 'restart' + <enter>.";
+    String message = "We are done. Please scan the following QR-code with the blue scanner.\n"
+            + "Then type 'exit' + <enter> or 'restart' + <enter>."
```
Comment (syncom) [inline java/gui/src/main/java/com/squareup/subzero/framebuffer/Screens.java:216]:
```suggestion
            + "Then type 'exit' + <enter> or 'restart' + <enter>.\n"
```
This is a nit.

**Comment by nmahendru** [comment]
IMG_5924
IMG_5923

Tested on my setup and the new screens look like above. 
It does shutdown the machine :)

**Review by syncom — APPROVED** [review]
:shipit:

**Review by oreparaz — APPROVED** [review]
thank you

**Pull request merged**